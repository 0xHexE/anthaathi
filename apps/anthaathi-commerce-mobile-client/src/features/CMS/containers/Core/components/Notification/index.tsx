import {View} from 'react-native';
import React from 'react';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';
import {Divider, Text} from 'react-native-paper';
import {CoreComponentType} from '../../../../types/common';

export interface NotificationProps {
  title: string;
  subtitle1: string;
  subtitle2: string;
  time: string;
  icon: string;
}

const Notification = ({
  title,
  subtitle1,
  subtitle2,
  time,
  icon,
}: NotificationProps) => {
  return (
    <View
      testID="notification"
      style={{
        // borderBottomWidth: 0.2,
        // borderBottomColor: '#364A15',
        marginVertical: 5,
        marginHorizontal: 10,
        padding: 5,
      }}>
      <View style={{flexDirection: 'row', justifyContent: 'space-between'}}>
        <View style={{flexDirection: 'row'}}>
          <SimpleLineIcons name={icon} size={18} />
          <Text
            style={{
              fontSize: 14,
              color: '#364A15',
              fontWeight: '400',
              marginLeft: 10,
            }}>
            {subtitle1}
          </Text>
        </View>
        <Text style={{fontSize: 14, color: '#364A15', fontWeight: '400'}}>
          {time}
        </Text>
      </View>
      <Text
        style={{
          color: '#364A15',
          fontSize: 16,
          fontWeight: '600',
          marginLeft: 28,
        }}>
        {title}
      </Text>
      <Text
        style={{
          fontSize: 14,
          color: '#364A15',
          fontWeight: '400',
          marginLeft: 28,
        }}>
        {subtitle2}
      </Text>
      <Divider bold={true} style={{marginTop: 10}} />
    </View>
  );
};

export default Notification;

export const NotificationCMSInput = {
  _component: CoreComponentType.Notification,
  component: Notification,
};
