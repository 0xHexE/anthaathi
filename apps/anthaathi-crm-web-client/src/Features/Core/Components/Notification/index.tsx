import React from 'react';
import { styled, useStyletron } from 'baseui';
import { Button, KIND, SIZE } from 'baseui/button';
import { LabelMedium, LabelSmall, LabelXSmall } from 'baseui/typography';
import { Block } from 'baseui/block';

export const NotificationBar = styled('div', () => ({
  position: 'sticky',
  maxHeight: '138px',
  margin: '12px',
  boxShadow: `rgba(0, 0, 0, 0.08) 0px 0.9px 4px, rgba(0, 0, 0, 0.06) 0px 2.6px 8px, rgba(0, 0, 0, 0.05) 0px 5.7px 12px, rgba(0, 0, 0, 0.04) 0px 15px 15px`,
  borderRadius: '6px',
  border: 'medium',
  width: '540px',
}));

export interface NotificationProps {
  id: string;
  time: string;
  type: 'Alert' | 'News' | 'Report'; // badge type
  badgeColor?: string;
  severity?: 'Critical' | 'Warning'; // badge alert :
  iconType?: React.ReactNode;
  isLoading?: string;
  title: string;
  buttonTitle?: string;
  buttonIcon: React.ReactNode;
  onClick1?: () => void;
  onClick2?: () => void;
  messages: string[];
  isRead: boolean;
}

function Notification({
  id,
  isLoading,
  time,
  type,
  severity,
  badgeColor,
  iconType,
  title,
  buttonTitle,
  buttonIcon,
  onClick1,
  onClick2,
  messages,
  isRead,
}: NotificationProps) {
  const [css, $theme] = useStyletron();
  return (
    <div>
      <NotificationBar>
        <Block padding="12px 8px 12px 8px">
          <Block display="flex" justifyContent="space-between">
            <Block display="flex" alignItems="center">
              {!isRead && (
                <Button
                  kind={KIND.secondary}
                  size={SIZE.compact}
                  $style={{
                    width: '24px',
                    height: '24px',
                    padding: '0px',
                    borderRadius: '4px',
                    backgroundColor: '#fff',
                    ':hover': { backgroundColor: $theme.colors.notificationA },
                  }}
                  onClick={onClick1}
                >
                  <span
                    className="fa fa-circle"
                    aria-hidden="true"
                    style={{ color: $theme.colors.notificationB, fontSize: 8 }}
                  />
                </Button>
              )}
              <div style={{ marginLeft: isRead ? '29px' : '5px' }}>
                {iconType}
              </div>
              <LabelXSmall
                className={[
                  css({
                    padding: '1px 8px 1px 8px',
                    borderRadius: '3px',
                    marginLeft: '5px',
                  }),
                  type === 'Report' &&
                    css({
                      borderColor: $theme.colors.notificationC,
                      border: '1px solid',
                      backgroundColor: '#fff',
                    }),
                  type === 'Alert' &&
                    css({
                      backgroundColor:
                        severity === 'Warning'
                          ? $theme.colors.notificationD
                          : $theme.colors.notificationE,
                    }),
                  type === 'News' &&
                    css({
                      backgroundColor: $theme.colors.notificationF,
                    }),
                  badgeColor &&
                    css({
                      backgroundColor: badgeColor,
                    }),
                ].join(' ')}
              >
                {type} {type === 'Alert' && severity ? ': ' + severity : null}
              </LabelXSmall>
            </Block>
            <Block display="flex" alignItems="center">
              <LabelSmall
                $style={{
                  color: $theme.colors.notificationG,
                }}
              >
                {time}
              </LabelSmall>
              <Button
                kind={KIND.secondary}
                size={SIZE.compact}
                $style={{
                  width: '24px',
                  height: '24px',
                  padding: '0px',
                  borderRadius: '4px',
                  backgroundColor: '#fff',
                  marginLeft: '5px',
                  ':hover': { backgroundColor: $theme.colors.notificationA },
                }}
                onClick={() => {}}
              >
                <span
                  className="fa fa-ellipsis-v"
                  aria-hidden="true"
                  style={{ color: $theme.colors.notificationH }}
                />
              </Button>
            </Block>
          </Block>
          <div style={{ marginLeft: '29px' }}>
            <LabelMedium
              $style={{
                fontFamily: $theme.typography.headingFontFamily,
                padding: '0px',
                margin: '0px',
                color: $theme.colors.notificationI,
                ':hover': {
                  textDecoration: 'underline',
                },
              }}
            >
              {title}
            </LabelMedium>
            {messages.map((data, index) => (
              <LabelSmall
                key={index}
                $style={{
                  color: $theme.colors.notificationJ,
                  marginTop: '5px',
                }}
              >
                {data}
              </LabelSmall>
            ))}
            <Button
              startEnhancer={buttonIcon}
              kind={KIND.tertiary}
              size={SIZE.compact}
              $style={{
                padding: '0px',
                borderRadius: '4px',
                backgroundColor: '#fff',
                margin: '10px 0 0 0',
                color: $theme.colors.notificationK,
                ':hover': {
                  backgroundColor: '#fff',
                  textDecoration: 'underline',
                },
              }}
              onClick={onClick2}
            >
              {buttonTitle}
            </Button>
          </div>
        </Block>
      </NotificationBar>
    </div>
  );
}

export default Notification;
