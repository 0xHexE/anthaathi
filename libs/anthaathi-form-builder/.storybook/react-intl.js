const locales = ['en-US'];

const messages = locales.reduce(
  (acc, lang) => ({
    ...acc,
    [lang]: import(`../locales/compiled-locales/${lang}.json`), // whatever the relative path to your messages json is
  }),
  {},
);

const formats = {};

export const reactIntl = {
  defaultLocale: 'en-US',
  locales,
  messages,
  formats,
};
