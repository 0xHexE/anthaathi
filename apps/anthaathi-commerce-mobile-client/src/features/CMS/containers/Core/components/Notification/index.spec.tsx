import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import Notification from '.';

describe('Notification', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <Notification
            title="title"
            subtitle1="subtitle1"
            subtitle2="subtitle2"
            time="2d"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('notification')).toBeTruthy();
  });

  it('should have Notification title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <Notification
            title="title"
            subtitle1="subtitle1"
            subtitle2="subtitle2"
            time="2d"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('title')).toBeTruthy();
    expect(temp.queryByText('subtitle1')).toBeTruthy();
    expect(temp.queryByText('subtitle2')).toBeTruthy();
    expect(temp.queryByText('2d')).toBeTruthy();
  });
});
