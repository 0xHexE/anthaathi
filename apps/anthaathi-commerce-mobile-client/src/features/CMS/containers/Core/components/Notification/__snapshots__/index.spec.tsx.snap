// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render the item 1`] = `
<View
  style={
    {
      "marginHorizontal": 10,
      "marginVertical": 5,
      "padding": 5,
    }
  }
  testID="notification"
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 18,
            },
            undefined,
            {
              "fontFamily": "simple-line-icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î€§
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "#364A15",
              "fontSize": 14,
              "fontWeight": "400",
              "marginLeft": 10,
            },
          ]
        }
      >
        subtitle1
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          {
            "color": "#364A15",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      2d
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        {
          "textAlign": "left",
        },
        {
          "color": "#364A15",
          "fontSize": 16,
          "fontWeight": "600",
          "marginLeft": 28,
        },
      ]
    }
  >
    title
  </Text>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        {
          "textAlign": "left",
        },
        {
          "color": "#364A15",
          "fontSize": 14,
          "fontWeight": "400",
          "marginLeft": 28,
        },
      ]
    }
  >
    subtitle2
  </Text>
  <View
    bold={true}
    style={
      [
        {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 0.5,
        },
        undefined,
        {
          "marginTop": 10,
        },
      ]
    }
  />
</View>
`;
